tools/bridge/bridge/confirmation_sender.py

```diff
@@ -0,0 +1,105 @@
+import logging
+from typing import Any, Dict
+
+import gevent
+from gevent.queue import Queue
+
+from web3.contract import Contract
+from eth_keys.datatypes import PrivateKey
+from eth_utils import keccak, int_to_big_endian
+
+from bridge.constants import HOME_CHAIN_STEP_INTERVAL
+
+
+class ConfirmationSender:
+    """Sends confirmTransfer transactions to the home bridge contract."""
+
+    def __init__(
+        self,
+        transfer_queue: Queue,
+        home_bridge_contract: Contract,
```

Comment:
Suggestion: I would rather not have the caller to take care about creating the contract instance. It is just complexity we don't need there. This makes even more sense if would like to continue to integrate some checks for this contract like we do for the event handler. So make sure that any contract is deployed at this address and at least make sure that the `confirmTransfer` function is there. We wanted to be better at such parts than the POA bridge. And I think this is super easy to implement, but saves a lot of possible trouble on getting a bridge validator running.
Anyways I would argue that we should try to stick with one pattern. Either the components/worker handle the blockchain components they connect to on their own or none of of them does.

Reply:
I agree that checks if the contract exists or not is useful. But to me this is validation of user input and should be done as early as possible, not here.

Reasons why I prefer passing the contract object instead of creating it in the class:
- we don't have to pass both address and ABI, so we save one argument
- it keeps the class smaller
- we might want to use the same contract objects in multiple places (can't think of an example right now though, but it doesn't seem unlikely)

Reply:
> and should be done as early as possible,

I though about it as well. My first idea was to integrate such check within the configuration validation. Unfortunately we need a web3 connection therefore already we don't have. So we would need a further module to provide such functionality. I would argue that our contract ABI constants could be that minimal, that testing for the signature hash of each entry should work out. So you could say, this is the contract, please check if the attached ABI is fully supported.

Reply:
[Here](https://github.com/trustlines-protocol/blockchain/pull/257) is a PR focusing this problem.